flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,531 @@
+import base64
+import os
+import time
+import urllib.parse
+
+from loguru import logger
+from requests import Response
+from requests.exceptions import RequestException
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.plugin import DependencyError, PluginError
+from flexget.task import Task
+
+try:
+    from jwt import decode
+
+    imported = True
+except ImportError:
+    imported = False
+
+logger = logger.bind(name='flood')
+
+
+class FloodClient:
+    success_list = [200, 202]
+
+    @staticmethod
+    def request(task: Task, config: dict, method: str, url: str, **kwargs) -> Response:
+        """
+        Send a request to Flood.
+        Returns None if the response status_code is not in the success_list.
+        """
+
+        try:
+            response = task.requests.request(
+                method, urllib.parse.urljoin(config["url"], url), **kwargs
+            )
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to send a request to Flood: {e}"
+            )
+
+        if not response.status_code in FloodClient.success_list:
+            raise PluginError(
+                f"Failed to send request to Flood. Received status {response.status_code} expected one of {FloodClient.success_list}."
+            )
+
+        return response
+
+    @staticmethod
+    def is_jwt_invalid(task: Task) -> bool:
```

Comment:
I'm not in love with negative conditionals like this. I'd probably make it `is_jwt_valid()` and return true for the positive case but I suppose that's a matter of taste.

Reply:
Yeah I don't like it either.. I'm not sure what was going through my head if I'm being completely honest. I think I just pressed F2 and went easy peasy without a second thought.